### PR TITLE
Cnh/adios circleci hello GitHub actions

### PR DESCRIPTION
**Pull request by christophernhill**
Changes to try switching to GitHub actions for blog generation

### CODE REVIEW

File: .github/workflows/update-posts.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    - name: Create conda environment
+      run: conda create --quiet -c conda-forge --name usrse_blog requests
+
+    - name: Update posts
+      run: |
+        echo "Looking for new posts..."
+        export PATH="/usr/share/miniconda/bin:$PATH"
+        source activate usrse_blog
+        pip install -r .github/requirements.txt
```
Comment (vsoch) [inline .github/workflows/update-posts.yaml:23]: Double check I didn’t move this path!

File: .github/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+python-frontmatter>=0.4.2
+feedparser>=5.2.1
+pyaml>=17.12.1
+# sgmllib not in python 3; workaround solution for feedparser relative link resolution
+sgmllib3k
+beautifulsoup4
```
Comment (vsoch) [inline .github/requirements.txt:6]: You shouldn’t need to update this file - the versions are the same and bs4 I added for some extended parsing (not used here!)

File: .github/workflows/update-posts.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      with:
+        ruby-version: 2.7
+
+    - name: Build site  
+      run: |
+        bundle install
+        bundle exec jekyll build
+
+    - name: Deploy
+      if: github.ref == 'refs/heads/main'
```
Comment (vsoch) [inline .github/workflows/update-posts.yaml:36]: This should probably be master

File: .github/workflows/update-posts.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        rm -rf _site
+        rm -rf .jekyll-cache
+        git add ./_posts/*       
+        git status
+        if git diff-index --quiet HEAD --; then
+            printf "No changes\n"
+        else
+           printf "Changes\n"
+           git commit -a -m "Automated deployment to update community posts $(date '+%Y-%m-%d')"
+           git push origin main
```
Comment (vsoch) [inline .github/workflows/update-posts.yaml:55]: Ditto master here!

**Review by vsoch — COMMENTED** [review]
Some quick comments!

File: .github/workflows/update-posts.yaml (hunk 4 above)
Comment (christophernhill) [reply .github/workflows/update-posts.yaml]: 👍

File: .github/workflows/update-posts.yaml (hunk 3 above)
Comment (christophernhill) [reply .github/workflows/update-posts.yaml]: 👍

File: .github/requirements.txt (hunk 2 above)
Comment (christophernhill) [reply .github/requirements.txt]: 👍 no soup 🥣

**Comment by vsoch** [comment]
Thank you @christophernhill for the speedy response! :rabbit:

**Comment by christophernhill** [comment]
Thanks @vsoch for the help. 
I'll delete paused .circleci before end of week and check everything still works. 

I may add Github action previews along lines of https://github.com/mit-orcd/orcd-docs-edit/blob/main/.github/workflows/pr-preview-artifact.yml to USRSE when I get a chance. 
Got to dash to something else for now.

**Comment by vsoch** [comment]
Dash away, dash away, dash away all! :dash:

**Comment by vsoch** [comment]
@christophernhill I've been loving using Netlify for hpc.social previews - I think with regular job updates it would be too much for the US-RSE site, and definitely here with a nightly push (it's like 300 build minutes max!) but you can add `[skip ci]` to any commit to try and more effectively use it, and even add automation to do that.

ssh preview to your own server is something I haven't tried - looks super cool! Can you ping me to take a look when you get it setup? It definitely would be nice to have something easier than the Circle setup, which has always been too-many-clicks, less than ideal.

**Pull request merged**